pkg/cli/other.go

```diff
@@ -0,0 +1,855 @@
+	}
+
+	// Get organization slug
+	org, err := promptForInput("Organization slug: ", false)
+	if err != nil {
+		return err
+	}
+	if org == "" {
+		return errors.New("organization slug cannot be empty")
+	}
+
+	// Check if token already exists for this organization
+	existingToken := f.Config.GetTokenForOrg(org)
+	if existingToken != "" {
+		fmt.Printf("Using existing API token for organization: %s\n", org)
+		return f.Config.SelectOrganization(org, f.GitRepository != nil)
+	}
+
+	// Get API token with password input (no echo)
+	token, err := promptForInput("API Token: ", true)
+	if err != nil {
+		return err
+	}
+	if token == "" {
+		return errors.New("API token cannot be empty")
+	}
+
+	fmt.Println("API token set for organization:", org)
+	return configureWithCredentials(f, org, token)
+}
+
+// promptForInput handles terminal input with optional password masking
+func promptForInput(prompt string, isPassword bool) (string, error) {
+	fmt.Print(prompt)
+
+	if isPassword {
+		// Use term.ReadPassword for secure password input
+		passwordBytes, err := term.ReadPassword(int(syscall.Stdin))
+		fmt.Println() // Add a newline after password input
+		if err != nil {
+			return "", err
+		}
+		return string(passwordBytes), nil
+	} else {
+		// Use standard input for regular text
+		reader := bufio.NewReader(os.Stdin)
+		input, err := reader.ReadString('\n')
+		if err != nil {
+			return "", err
+		}
+		// Trim whitespace and newlines
+		return strings.TrimSpace(input), nil
+	}
+}
+
+func validateConfig(conf *config.Config) error {
+	if conf.APIToken() == "" {
+		return bkErrors.NewConfigurationError(nil, "API token not configured. run `bk configure` to set it up")
+	}
+	if conf.OrganizationSlug() == "" {
+		return fmt.Errorf("no organization selected. run `bk use` to select one")
+	}
+	return nil
+}
+
+// parseBuildIdentifier parses a build identifier which can be:
+// - A build URL (e.g., "https://buildkite.com/org/pipeline/builds/123")
+// - For now, build numbers without context are not supported
+func parseBuildIdentifier(identifier, defaultOrg string) (org, pipeline, buildNumber string, err error) {
+	// If it looks like a URL, parse it
+	if strings.HasPrefix(identifier, "http") {
+		u, parseErr := url.Parse(identifier)
+		if parseErr != nil {
+			return "", "", "", fmt.Errorf("invalid build URL: %w", parseErr)
+		}
+
+		// Expected format: /org/pipeline/builds/number
+		parts := strings.Split(strings.Trim(u.Path, "/"), "/")
+		if len(parts) >= 4 && parts[2] == "builds" {
+			return parts[0], parts[1], parts[3], nil
+		}
+		return "", "", "", fmt.Errorf("invalid build URL format")
+	}
+
+	// Build numbers without pipeline context require more complex resolution
+	// For now, require full URLs for simplicity
+	return "", "", "", fmt.Errorf("build number requires pipeline context - please use full build URL like: https://buildkite.com/org/pipeline/builds/%s", identifier)
```

Comment:
We could previously use `bk build view $NUMBER`, now it expects full URI. Why's that?